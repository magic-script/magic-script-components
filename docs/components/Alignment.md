---
id: alignment
title: Alignment Type
---

# Alignment Type options:
 - top-left
 - top-center
 - top-right
 - center-left
 - center-center
 - center-right
 - bottom-left
 - bottom-center
 - bottom-right
