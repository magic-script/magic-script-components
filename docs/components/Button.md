---
id: button
title: Button
---

## Description

Button represents a standard, clickable button that you can select and activate with the Control. You can set the label text and control the size and color of the button or use an icon. Button text uses the default system font, Lomino, and is white.

## Example

```javascript
class MyApp extends React.Component {
  buttonClickHandler = event => {
    console.log("Hello from buttonClickHandler");
  };

  render() {
    return (
      <View>
        <Button
          height={0.1}
          localPosition={[0.4, -0.2, 0]}
          onClick={this.buttonClickHandler}
          roundness={0.7}
          textSize={0.03}
          width={0.3}
        >
          Click Me
        </Button>
      </View>
    );
  }
}
```

## [Common Events](../types/Events.md)

## [Common Properties](../types/Properties.md)

## Create Properties

| Name      | Type   | Default Value | Description                                                                                          |
| --------- | ------ | :-----------: | ---------------------------------------------------------------------------------------------------- |
| text      | string |      n/a      | The UTF-8 encoded text to set the initial value of the button label.                                       |
| width     | number |      `0`      | The width of the button, 0 for default.                                                              |
| height    | number |      `0`      | The height of the button, 0 for default.                                                             |
| roundness | number |      1.0      | The ratio of corner rounding (0-1). Use `1` for a fully rounded button, such as a circle or capsule. Use `0` for a sharp corner. |

## Element Properties

| Name      | Type   | Description                               |
| :-------- | :----- | ----------------------------------------- |
| iconColor | vec4   | Sets the RGBA color of the button icon.   |
| iconSize  | vec2   | Sets the button icon size in scene units. |
| text      | string | Sets the UTF-8 encoded button text.       |
| textColor | vec4   | Sets the RGBA color of the button text.   |
| textSize  | number | Sets the text size.                       |
