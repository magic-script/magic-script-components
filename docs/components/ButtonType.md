---
id: buttonType
title: Button Type
---

# Button Type options:
 - icon
 - icon-with-label
 - text
 - text-with-icon
