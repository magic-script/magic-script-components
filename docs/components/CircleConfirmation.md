---
id: circleConfirmation
title: CircleConfirmation
---

import PlatformTabs from '@theme/PlatformTabs';

<PlatformTabs component='circleconfirmation' />â€‹

## Description

CircleConfirmation is the UI element intended to confirm powering off. The user moves the control knob using the touchpad of the Control. When the user makes a complete circle with the control knob, CircleConfirmation signals this to the client code. The client code can then power off upon confirmation receipt.

## Example

```javascript
import React from 'react';
import { View, CircleConfirmation } from 'magic-script-components';

export default class MyApp extends React.Component {
    constructor(props) {
        super(props);
    }

    onConfirmationCanceled = event => {
        console.log("onConfirmationCanceled");
    };
    onConfirmationUpdate = event => {
        console.log("onConfirmationUpdate");
    };
    onConfirmationComplete = event => {
        console.log("onConfirmationComplete");
    };


    render() {
        return (
            <View>
                <CircleConfirmation
                    onConfirmationCanceled={this.onConfirmationCanceled}
                    onConfirmationCompleted={this.onConfirmationComplete}
                    onConfirmationUpdated={this.onConfirmationUpdate}
                    radius={0.2}
                ></CircleConfirmation>
            </View>
        );
    }
}
```

## [Common Events](../events/CommonEvents.md)

## [Common Properties](../types/Properties.md)

## Create Properties

| Name   | Type   | Default Value | Required | Description                                                                                                                                                                       |
| ------ | ------ |    :------:   | :------: |---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| radius | number |      0.2      |    N     |The radius of the circle the control knob travels to complete the confirmation. The default height is 0.1 (100mm). Setting the value to '0' sets the height to the default height. |
