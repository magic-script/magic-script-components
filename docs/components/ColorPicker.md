---
id: colorPicker
title: ColorPicker
---

import PlatformTabs from '@theme/PlatformTabs';

<PlatformTabs component='colorpicker' />‚Äã


## Description

Dialog which allows the user to select color.

## Example

```javascript
import React from 'react';
import { View, ColorPicker } from 'magic-script-components';

export default class MyApp extends React.Component {
    onColorChanged = event => {
        // event.Color
        console.log("onColorChanged event received: ", event);
    };

    onColorConfirmed = event => {
        // event.Color
        console.log("onColorConfirmed event received: ", event);
    };

    onColorCanceled = event => {
        // event.Color
        console.log("onColorCanceled event received: ", event);
    };

    render() {
        return (
            <View name="main-view">
                <ColorPicker
                    color={[1, 0.5, 1, 0.8]}
                    onColorChanged={this.onColorChanged}
                    onColorConfirmed={this.onColorConfirmed}
                    onColorCanceled={this.onColorCanceled} />
            </View>
        );
    }
}
```

## [Common Events](../events/CommonEvents.md)

## [Common Properties](../types/Properties.md)

## Create Properties
| Name   | Type   |    Default Value     | Required | Description                             | Lumin Os |  IOS  | Android |
| ------ | ------ | :------------------: | :------: | --------------------------------------- | :------: | :---: | :-----: |
| color  | string | [1.0, 1.0, 1.0, 1.0] |    N     | The initial color for the color picker. |    üëç    |  üëç   |   üëç    |
| height | number |          0           |    N     | Optional height for the color picker.   |    üëç    |       |         |

## Element Properties

| Name  | Type   |    Default Value     | Description                             |
| :---- | :----- | :------------------: | --------------------------------------- |
| color | string | [1.0, 1.0, 1.0, 1.0] | The initial color for the color picker. |

## Component specific events
| Name             | Event Field     | Description                              |
| :--------------- | :-------------- | ---------------------------------------- |
| onColorChanged   | `Color: string` | Notifiy when color in picker has changed |
| onColorConfirmed | `Color: string` | Notify when user confirm selected color  |
| onColorCanceled  | `Color: string` | Notify when user cancel selected color   |