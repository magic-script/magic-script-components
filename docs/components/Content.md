---
id: content
title: Content
---

## Tag: `<Content>`

- #### Lumin Runtime native component - TransformNode

## [Common Events](../types/Events.md)

## [Common Properties](../types/Properties.md)

## Create Properties

| Name            | Type   |    Default Value     | Description                             |
| --------------- | ------ | :------------------: | --------------------------------------- |
| localTransform  |  mat4  | 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1 | The local transform of the new node. Use glm::mat4() for the identity matrix. |

### Purpose:
Some components need to wrap their children in `<content>` tag.
Example:
```javascript
<ScrollView>
    <ScrollBar />
    <Content>
        <Text>Message Box</Text>
        <Button>Submit</Button>
    </Content>
</ScrollView>
```
