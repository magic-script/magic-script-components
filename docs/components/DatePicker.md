---
id: datePicker
title: DatePicker
---

## Example

```javascript
<DatePicker
    label='This is label'
    labelSide='left'
    height={0.5}
    defaultDate='03/21/1990'
    yearMin={1990}
    yearMax={2100}
    localScale={[2, 2, 0]}
/>
```

## [Common Events](../types/Events.md)

## [Common Properties](../types/Properties.md)

## Create Properties

| Name        | Type   |   Default Value    | Description                                                                                               |
| ----------- | ------ | :----------------: | --------------------------------------------------------------------------------------------------------- |
| label       | string |        n/a         | The text label that appears with the DatePicker.                                                          |
| labelSide   | string |       `top`        | Side enum specifying whether the label appears on the side or top. Only top and left currently supported. |
| defaultDate | string | (current DateTime) | The default date to appear on the picker instead of the placeholder strings.                              |
| yearMin     | number |        `-1`        | The minimum year available to choose from.                                                                |
| yearMax     | number |        `-1`        | The maximum year available to choose from.                                                                |

### LabelSide values:

- bottom
- left
- right
- top

## Element Properties

| Name  | Type   | Description                                                                       |
| :---- | :----- | --------------------------------------------------------------------------------- |
| color | string | Sets the DatePicker component's color, doesn't affect label.                      |
| date  | number | Sets the Date manually. This will only have an affect when not currently focused. |
