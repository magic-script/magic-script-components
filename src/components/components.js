import React from 'react'

export { Button } from './button';

export function Model (props) {
    // return (<model {...props} />);
    return React.createElement('model', props);
}

export function Slider (props) {
    // return (<slider {...props} />);
    return React.createElement('slider', props);
}

export { Text } from './text';

export function View (props) {
    // return (<view {...props} />);
    return React.createElement('view', props);
}

export function Toggle (props) {
    return React.createElement('toggle', props);
}

export function ToggleGroup (props) {
    // return (<toggleGroup {...props} />);
    return React.createElement('toggleGroup', props);
}

export function TextEdit (props) {
    // return (<textEdit {...props} />);
    return React.createElement('textEdit', props);
}

export function Image (props) {
    // return (<image {...props} />);
    return React.createElement('image', props);
}

export function ScrollBar (props) {
    // return (<scrollBar {...props} />);
    return React.createElement('scrollBar', props);
}

export function ScrollView (props) {
    // return (<scrollView {...props} />);
    return React.createElement('scrollView', props);
}

export function ListView (props) {
    // return (<listView {...props} />);
    return React.createElement('listView', props);
}

export function ListViewItem (props) {
    // return (<listViewItem {...props} />);
    return React.createElement('listViewItem', props);
}

export function Spinner (props) {
    // return (<spinner {...props} />);
    return React.createElement('spinner', props);
}

export function ProgressBar (props) {
    // return (<progressBar {...props} />);
    return React.createElement('progressBar', props);
}

export function GridLayout (props) {
    // return (<gridLayout {...props} />);
    return React.createElement('gridLayout', props);
}

export function LinearLayout (props) {
    // return (<linearLayout {...props} />);
    return React.createElement('linearLayout', props);
}

export function RectLayout (props) {
    // return (<rectLayout {...props} />);
    return React.createElement('rectLayout', props);
}

export function DropdownList (props) {
    // return (<dropdownList {...props} />);
    return React.createElement('dropdownList', props);
}

export function DropdownListItem (props) {
    // return (<dropdownListItem {...props} />);
    return React.createElement('dropdownListItem', props);
}

export function Panel (props) {
    // return (<panel {...props} />);
    return React.createElement('panel', props);
}

export function Tab (props) {
    // return (<tab {...props} />);
    return React.createElement('tab', props);
}

export function Dialog (props) {
    // return (<dialog {...props} />);
    return React.createElement('dialog', props);
}

export function PageView (props) {
    // return (<pageView {...props} />);
    return React.createElement('pageView', props);
}

export function WebView (props) {
    // return (<webView {...props} />);
    return React.createElement('webView', props);
}

export function PortalIcon (props) {
    // return (<portalIcon {...props} />);
    return React.createElement('portalIcon', props);
}

export function ColorPicker (props) {
    // return (<colorPicker {...props} />);
    return React.createElement('colorPicker', props);
}

export function TimePicker (props) {
    // return (<timePicker {...props} />);
    return React.createElement('timePicker', props);
}

export function DatePicker (props) {
    // return (<datePicker {...props} />);
    return React.createElement('datePicker', props);
}

export function CircleConfirmation (props) {
    // return (<circleConfirmation {...props} />);
    return React.createElement('circleConfirmation', props);
}

export function Content (props) {
    // return (<content {...props} />);
    return React.createElement('content', props);
}

export function Quad (props) {
    // return (<quad {...props} />);
    return React.createElement('quad', props);
}

export function Video (props) {
    // return (<video {...props} />);
    return React.createElement('video', props);
}

export function Audio (props) {
    // return (<audio {...props} />);
    return React.createElement('audio', props);
}

export function Light (props) {
    // return (<light {...props} />);
    return React.createElement('light', props);
}

export function Line (props) {
    // return (<line {...props} />);
    return React.createElement('line', props);
}

export function Prism (props) {
    // return (<prism {...props} />);
    return React.createElement('prism', props);
}

export function Scene (props) {
    // return (<scene {...props} />);
    return React.createElement('scene', props);
}
