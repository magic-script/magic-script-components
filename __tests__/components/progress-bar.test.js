// Copyright (c) 2020 Magic Leap, Inc. All Rights Reserved

import React from "react";
import renderer from "react-test-renderer";

import { ProgressBar } from "../../src/components/components.js";

describe("ProgressBar component", () => {
  test("should contain mapped properties", () => {
  const progressColor = {
    beginColor: "white",
    endColor: "blue",
  };

  const component = renderer.create(
    <ProgressBar
      beginColor={progressColor.beginColor}
      endColor={progressColor.endColor}
    />
  );

  const json = component.toJSON();
  expect(json).toMatchSnapshot();

  expect(json.props).toHaveProperty(
    "progressColor.beginColor",
    progressColor.beginColor
  );
  expect(json.props).toHaveProperty(
    "progressColor.endColor",
    progressColor.endColor
  );
});
});
