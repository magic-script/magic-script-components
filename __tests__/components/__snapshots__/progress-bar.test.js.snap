// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar component should contain mapped properties 1`] = `
<progressBar
  alignment="center-center"
  debug={false}
  enabled={true}
  height={0}
  hideBounds={false}
  localPosition={
    Array [
      0,
      0,
      0,
    ]
  }
  localRotation={
    Array [
      0,
      0,
      0,
      1,
    ]
  }
  localScale={
    Array [
      1,
      1,
      1,
    ]
  }
  max={1}
  min={0}
  progressColor={
    Object {
      "beginColor": "white",
      "endColor": "blue",
    }
  }
  skipRaycast={false}
  value={0}
  visible={true}
  width={0.5}
/>
`;
